src/feature.js

```diff
@@ -646,17 +648,24 @@ var feature = function (arg) {
    * Get/Set bin of the feature.  The bin number is used to determine the order
    * of multiple features on the same layer.  It has no effect except on the
    * vgl renderer.  A negative value hides the feature without stopping
-   * interaction with it.  Otherwise, more features with higher bin numbers are
+   * interaction with it.  Otherwise, features with higher bin numbers are
    * drawn above those with lower bin numbers.  If two features have the same
    * bin number, their order relative to one another is indeterminate and may
    * be unstable.
    *
    * @param {number} [val] The new bin number.  If `undefined`, return the
-   *    current bin number.
+   *    current bin number.  If `null`, the bin is dynamically computed based
+   *    on order within the parent.  If children are nested, this may not be
+   *    what is desired.
    * @returns {number|this} The current bin number or a reference to `this`.
    */
   this.bin = function (val) {
     if (val === undefined) {
+      if (m_bin === null) {
+        var parent = this.parent(),
+            idx = parent ? parent.children().indexOf(m_this) : -1;
```

Comment:
why not just set it to 0 (instead of -1) here. Also, if we are returning a value even if m_bin is null, should we store this value in m_bin before returning it?

Reply:
`indexOf` returns -1 if the value isn't present, so return -1 here results in the same result if there is no parent or this is somehow not a child of its own parent.  Although we then clamp it to zero, I felt handling the missing parent case identically to a missing entry case was clearer then doing otherwise.

No, we shouldn't store the value in `m_bin`.  `null` is the special meaning of return the position based on where it is in the child list.  If you had two children, A, B, both with null m_bin, they would have bin values of 0, 1.  If you inserted a child, so you have A, C, B, they would then have bin values of 0, 1, 2 (so B would go from 1 to 2).  This is desired, and overriding this would break that feature.

Reply:
thanks makes sense to me, given this explanation.